src/index.js

```diff
@@ -86,13 +86,40 @@ exports.init = ({ bucket, mock }) => {
       return fs.createReadStream(localPath);
     };
 
-    // FIX: Upload to GCS once stream is closed.
-    exports.createWriteStream = async (remotePath) => {
+    exports.createWriteStream = (remotePath, contentType) => {
       let localPath = local(remotePath);
       ensureLocalPathDir(localPath);
-      return fs.createWriteStream(localPath);
+      
+      return storage.bucket(bucket).file(remotePath).createWriteStream({
+        metadata: {
+          contentType: contentType,
+        },
+      });
     };
 
+    exports.signedURLWrite = async (remotePath, data, contentType) => {
+
+      const options = {
+        action: 'write',
+        expires: Date.now() + 1 * 60 * 1000, // 1 minute
```

Comment:
expiry will be passed as a parameter by the caller. default expiry can be set to 10 minutes.